go-controller/pkg/ovn/master.go

```diff
@@ -844,28 +850,71 @@ func (oc *Controller) addNodeAnnotations(node *kapi.Node, hostSubnets []*net.IPN
 
 func (oc *Controller) addNode(node *kapi.Node) ([]*net.IPNet, error) {
 	oc.clearInitialNodeNetworkUnavailableCondition(node, nil)
-
-	hostSubnets, _ := util.ParseNodeHostSubnetAnnotation(node)
-	if hostSubnets != nil {
-		// Node already has subnet assigned; ensure its logical network is set up
-		return hostSubnets, oc.ensureNodeLogicalNetwork(node.Name, hostSubnets)
-	}
-
-	// Node doesn't have a subnet assigned; reserve a new one for it
-	hostSubnets, err := oc.masterSubnetAllocator.AllocateNetworks()
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node)
 	if err != nil {
-		return nil, fmt.Errorf("error allocating network for node %s: %v", node.Name, err)
-	}
-	klog.Infof("Allocated node %s HostSubnet %s", node.Name, util.JoinIPNets(hostSubnets, ","))
+		// Log the error and try to allocate new subnets
+		klog.Infof("Failed to get node %s host subnets annotations: %v", node.Name, err)
+	}
+
+	// OVN can work in single-stack or dual-stack only.
+	expectedHostSubnets := 1
+	// if dual-stack mode we expect one subnet per each IP family
+	if config.IPv4Mode && config.IPv6Mode {
+		expectedHostSubnets = 2
+	}
+	if expectedHostSubnets < len(hostSubnets) {
+		klog.Fatalf("Incompatible number of subnets on node %s, subnets assigned %v but only %d subnets allowed", node.Name, hostSubnets, expectedHostSubnets)
```

Comment:
fatal, you have a node annotated with let´s say 3 subnet and you only can have 2 or 1, this will require manual intervention to fix the node annotations IIUIC